A/AMReX/build_tarballs.jl

```diff
@@ -0,0 +1,69 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "AMReX"
+version = v"21.4.0"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/AMReX-Codes/amrex/releases/download/21.04/amrex-21.04.tar.gz", "1c610e4b0800b16f7f1da74193ff11af0abfb12198b36a7e565a6a7f793087fa")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd amrex
+mkdir build
+cd build
+if [[ "$target" == *-x86_64-w64-mingw32 ]]; then
+    # TODO: This case should not be necessary
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI -DMPI_C_LIBRARIES=msmpi64 -DMPI_CXX_LIBRARIES=msmpi64"
+elif [[ "$target" == *-mingw* ]]; then
+    mpiopts="-DMPI_HOME=$WORKSPACE/destdir -DMPI_GUESS_LIBRARY_NAME=MSMPI"
+else
+    mpiopts=
+fi
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_gcc.cmake -DCMAKE_BUILD_TYPE=Release -DAMReX_FORTRAN=OFF -DAMReX_OMP=ON -DAMReX_PARTICLES=ON -DBUILD_SHARED_LIBS=ON ${mpiopts} ..
+make -j$(nproc)
+make -j$(nproc) install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+# - We can't build with musl since AMReX requires the `fegetexcept` GNU API
+# - We might not want to build `cxxstring_abi="cxx03"` since AMReX requires at least C++14 (C++17 on Windows)
+platforms = [
+    Platform("aarch64", "linux"; libc="glibc"),
+    Platform("armv7l", "linux"; call_abi="eabihf", libc="glibc"),
+    Platform("armv7l", "linux"; libc="glibc"),
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("i686", "windows"),
+    Platform("powerpc64le", "linux"; libc="glibc"),
+    Platform("x86_64", "freebsd"),
+    Platform("x86_64", "linux"; libc="glibc"),
+    Platform("x86_64", "macos"),
+    Platform("x86_64", "windows"), # MPI_CXX not found
+]
+platforms = expand_cxxstring_abis(platforms)
```

Comment:
I'm not sure why, but we get the warning only for FreeBSD, the library for macOS doesn't seem to expose `std::string` symbols
```suggestion
platforms = expand_cxxstring_abis(platforms; skip=Sys.isapple)
```